contrib/intel/jenkins/summary.py

```diff
@@ -7,389 +8,493 @@
+                self.line = self.log.readline().lower()
 
-        if 'name:' in line:
-            test_name = line.split()[2:]
-            test_name_string = ' '.join(test_name)
+        if 'name:' in self.line:
+            test_name = self.line.split()[2:]
+            self.test_name_string = ' '.join(test_name)
 
-        if 'result:' in line:
-            result_line = line.split()
+    def get_result_line(self):
+        if 'result:' in self.line:
+            self.result_line = self.line.split()
             # lines can look like 'result: Pass' or
             # 'Ending test 1 result: Success'
-            result = (result_line[result_line.index('result:') + 1]).lower()
-            if result == 'pass' or result == 'success':
-                    passes += 1
-
-            if result == 'fail':
-                fails += 1
-                if 'ubertest' in test_name_string:
-                    idx = (result_line.index('result:') - 1)
-                    ubertest_number = int((result_line[idx].split(',')[0]))
-                    failed_tests.append(f"{test_name_string}: " \
-                                        f"{ubertest_number}")
-                else:
-                    failed_tests.append(test_name_string)
-
-            if result == 'excluded' or result == 'notrun':
-                excludes += 1
-                excluded_tests.append(test_name_string)
-
-        if "exiting with" in line:
-            fails += 1
-            failed_tests.append(test_name_string)
-
-        line = log.readline()
-
-    print_results(f"{prov} fabtests {build_mode}", passes, fails, failed_tests,
-                  excludes, excluded_tests)
-
-    log.close()
-    return int(fails)
+            self.result = (self.result_line[self.result_line.index(
+                           'result:') + 1]).lower()
+
+    def check_pass(self):
+        self.get_result_line()
+        if self.result == 'pass' or self.result == 'success':
+            self.passes += 1
+
+        self.result = ''
+
+    def check_fail(self):
+        self.get_result_line()
+        if self.result == 'fail':
+            self.fails += 1
+            if 'ubertest' in self.test_name_string:
+                idx = (self.result_line.index('result:') - 1)
+                ubertest_number = int((self.result_line[idx].split(',')[0]))
+                self.failed_tests.append(f"{self.test_name_string}: " \
+                                    f"{ubertest_number}")
+            else:
+                self.failed_tests.append(self.test_name_string)
+
+        if "exiting with" in self.line:
+            self.fails += 1
+            self.failed_tests.append(self.test_name_string)
+
+        self.result = ''
+
+    def check_exclude(self):
+        self.get_result_line()
+        if self.result == 'excluded' or self.result == 'notrun':
+            self.excludes += 1
+            self.excluded_tests.append(self.test_name_string)
+
+        self.result = ''
+
+class ZeSummarizer(Summarizer):
```

Comment:
ZeSummarizer seems to closely match FabtestsSummarizer.  It could be good to try and reduce some duplication.  The same comments I made for FabtestsSummarizer apply here, too